agent/agent.go

```diff
@@ -708,17 +713,20 @@ func (a *Agent) StopInputPlugin(input string) {
 
 	if !ok {
 		log.Printf("E! [agent] You are trying to stop an input that is not running: %s \n", input)
-		return
+		return fmt.Errorf("you are trying to stop an input that is not running")
 	}
 
-	statusChannel <- Stop
+	statusChannel <- struct{}{}
+	<-statusChannel
```

Comment:
Why do we need to wait for this channel again? Why can't we just send a signal to stop the gatherloop and then remove it immediately, even before that returns? 

This is super janky and won't work correctly in random scenarios.

Eg: sometimes you send a struct through statusChannel, and then the next line of this function is run before the gatherLoop is run, and then it sees something in the channel on line 720, and then passes through the function, however the gatherloop never gets the struct through the channel. So this function removes this input from the inputs but never actually stops the gatherloop.